mobius-coroutines/src/main/java/com/spotify/mobius/coroutines/CoroutinesSubtypeEffectHandlerBuilder.kt

```diff
@@ -23,68 +30,152 @@ import kotlin.reflect.KClass
  * instances of the builder are mutable and not thread-safe.
  */
 class CoroutinesSubtypeEffectHandlerBuilder<F : Any, E : Any> {
-    private val effectHandlersMap = mutableMapOf<KClass<out F>, suspend (F) -> Flow<E>>()
+    private val effectHandlersMap = mutableMapOf<KClass<out F>, EffectHandler<F, E>>()
 
     inline fun <reified G : F> addAction(
+        cancellationPolicy: CancellationPolicy = RunSequentially,
         crossinline action: suspend () -> Unit
-    ) = addFlowProducer<G> {
+    ) = addEffectHandler<G>(cancellationPolicy) { _, _ ->
         action.invoke()
-        flowOf()
     }
 
     inline fun <reified G : F> addConsumer(
```

Comment:
Missing documentation for the 'add*' methods, but I suppose that was already the case on master, so maybe a task for a separate PR.

Reply:
I added the documentation in this one.